app/services/edge_cache/bust.rb

```diff
@@ -1,5 +1,12 @@
 module EdgeCache
   class Bust
+    TIMEFRAMES = [
+      [1.week.ago, "week"],
```

Comment:
**bug (blocking):** This will not work as expected, as the timeframes will be relative to application start (when the class body gets evaluated), not the current time.

```ruby
TIMEFRAMES = [[1.week.ago, "week"]]
#=> [[Tue, 01 Dec 2020 05:35:36 UTC +00:00, "week"]]
```

I suggest making them lambdas instead and calling them in the code:

```ruby
TIMEFRAMES = [
  [-> { 1.week.ago }, "week"],
  ...
]

# e.g. app/services/edge_cache/bust_article.rb
next unless Article.published.where("published_at > ?", timestamp.call)
```

Reply:
Great call, updated!

Reply:
> the timeframes will be relative to application start (when the class body gets evaluated), not the current time...I suggest making them lambdas instead

Interesting; @citizen428 I wouldn't have thought to do this, TIL!

Reply:
@vaidehijoshi It's also a common problem in migrations, when people add new date columns and want to set a default: if you don't specify it in the correct way, the default will be the timestamp of the migration run, not evaluated on record creation.